packages/agreement-process/test/testUpgradeAgreement.ts

```diff
@@ -0,0 +1,1273 @@
+        version !== undefined
+          ? await readAgreementEventByVersion(agreementId, version, postgresDB)
+          : await readLastAgreementEvent(agreementId, postgresDB);
+
+      if (!actualAgreementData) {
+        fail("Creation fails: agreement not found in event-store");
+      }
+
+      expect(actualAgreementData).toMatchObject({
+        ...expectedStoreEvent,
+        event_version: 1,
+      });
+
+      return decoderFunction(actualAgreementData, expectedStoreEvent);
+    }
+
+    function toAgreementStampsV1(agreement: Agreement): StampsV1 {
+      const stamps: StampsV1 = {};
+
+      if (agreement.stamps.submission) {
+        stamps.submission = toStampV1(agreement.stamps.submission);
+      }
+
+      if (agreement.stamps.activation) {
+        stamps.activation = toStampV1(agreement.stamps.activation);
+      }
+
+      if (agreement.stamps.rejection) {
+        stamps.rejection = toStampV1(agreement.stamps.rejection);
+      }
+
+      if (agreement.stamps.suspensionByProducer) {
+        stamps.suspensionByProducer = toStampV1(
+          agreement.stamps.suspensionByProducer
+        );
+      }
+
+      if (agreement.stamps.suspensionByConsumer) {
+        stamps.suspensionByConsumer = toStampV1(
+          agreement.stamps.suspensionByConsumer
+        );
+      }
+
+      if (agreement.stamps.upgrade) {
+        stamps.upgrade = toStampV1(agreement.stamps.upgrade);
+      }
+
+      if (agreement.stamps.archiving) {
+        stamps.archiving = toStampV1(agreement.stamps.archiving);
+      }
+
+      return stamps;
+    }
+
+    async function uploadDocument(path: string, name: string): Promise<void> {
+      await fileManager.storeBytes(
+        config.s3Bucket,
+        config.consumerDocumentsPath,
+        path,
+        name,
+        Buffer.from("large-document-file")
+      );
+      expect(await fileManager.listFiles(config.s3Bucket)).toContainEqual(
+        `${config.consumerDocumentsPath}/${path}/${name}`
+      );
+    }
+
+    it("should succeed with valid Verified and Declared attributes", async () => {
+      const authData = {
+        ...getRandomAuthData(),
+        userRoles: [userRoles.INTERNAL_ROLE],
+      };
+      const tenantId = authData.organizationId.toString();
+
+      const validVerifiedTenantAttribute = getMockVerifiedTenantAttribute(
+        unsafeBrandId<AttributeId>(tenantId)
+      );
+      const validVerifiedEserviceAttribute = getMockEServiceAttribute(
+        validVerifiedTenantAttribute.id
+      );
+
+      const validDeclaredTenantAttribute = getMockDeclaredTenantAttribute(
+        unsafeBrandId<AttributeId>(tenantId)
+      );
+      const validDeclaredEserviceAttribute = getMockEServiceAttribute(
+        validDeclaredTenantAttribute.id
+      );
+
+      const descriptorId = generateId<DescriptorId>();
+      const deprecatedDescriptor = {
+        ...getMockDescriptorPublished(
+          descriptorId,
+          [],
```

Comment:
In this way, you are not explicitly covering the `validateCertifiedAttributes` part of the business logic:

<img width="523" alt="image" src="https://github.com/pagopa/interop-be-monorepo/assets/6418684/f3d458cc-2c2e-41c1-a6f8-87be5249e6d1">

If the requester is different from the `producerId`, it also checks for certified attributes validity - it is probably working because the certified attributes are empty, I would add them to include that in the success case

Reply:
make sense

Reply:
Left a comment where you create the valid certified attributes, about missing `revocationTimestamp` override